x-pack/plugins/maps/public/classes/sources/es_source/es_source.ts

```diff
@@ -291,15 +392,25 @@ export class AbstractESSource extends AbstractVectorSource {
     registerCancelCallback,
     sourceQuery,
     timeFilters,
-  }) {
+  }: {
+    layerName: string;
+    style: IVectorStyle;
+    dynamicStyleProps: Array<IDynamicStyleProperty<DynamicStylePropertyOptions>>;
+    registerCancelCallback: (callback: () => void) => void;
+    sourceQuery?: MapQuery;
+    timeFilters: TimeRange;
+  }): Promise<object> {
     const promises = dynamicStyleProps.map((dynamicStyleProp) => {
       return dynamicStyleProp.getFieldMetaRequest();
     });
 
     const fieldAggRequests = await Promise.all(promises);
-    const aggs = fieldAggRequests.reduce((aggs, fieldAggRequest) => {
-      return fieldAggRequest ? { ...aggs, ...fieldAggRequest } : aggs;
-    }, {});
+    const aggs: Record<string, any> = fieldAggRequests.reduce(
+      (aAggs: Record<string, any>, fieldAggRequest: unknown) => {
```

Comment:
`getFieldMetaRequest` can return null so `fieldAggRequest` should be typed as `unknown | null`